### DISCUSSION TITLE
Discord In-Game Overlay causes stutters

### DISCUSSION THREAD

**Discussion by Qerewe** [ppy/osu#24043] [Q&A]
I found out this was a Discord problem rather than an osu problem but I'm still going to make a discussion so this so it can be referenced in case anyone else has a similar issue and since I already brought it up in the Discord server.

### **Issue**

Having Discord in the background with in-game overlay enabled, the game's Update and Draw (as seen with crtl-F11) stutters when the Threading Mode is set to multithreaded. Switching to single thread fixes the stuttering (as does completely closing discord or turning off game overlay)

### **Solution**
After some more troubleshooting, I figured out it was a Discord problem as it happens with other games. Whenever I move the mouse while the game overlay is turned on but not showing, it raises my CPU usage to the point of bottlenecking my CPU (i3-9100F). The reason that setting it to single thread fixed it is because it lowers osu's CPU usage enough that it doesn't bottleneck when Discord raises it's usage. 

In-game Overlay Off
![Taskmgr_9jYymFGGst](https://github.com/ppy/osu/assets/20256930/75921dd3-76ec-45ea-9f4a-4fe3279ca248)

In-Game Overlay On and waving mouse around
![Taskmgr_FvzByzgRxr](https://github.com/ppy/osu/assets/20256930/919affe9-3367-447d-b96a-85baf7db7c10)

**Comment by peppy** [comment]
Does the scenario improve if you reduce the osu! frame limiter to something sane? That's probably the difference between multi/single threading.

I hypothesise that as you make the game render more frames, the discord overhead increase, as it's hooking on a per-frame basis.

**Reply by Qerewe** [reply]
The frame limiter was always using VSync, so it was always at 144hz with my monitor. For the sake of troubleshooting, I lowered my monitor's refresh rate to 60 and it did reduce the problem, but it would still peak and stutter every once in a while, just less frequently.
